framework/tlvf/src/src/CmduMessage.cpp

```diff
@@ -65,3 +65,12 @@ uint16_t CmduMessage::getMessageId()
 
     return mid;
 }
+
+void CmduMessage::setMessageId(uint16_t mid)
+{
+    auto cmduhdr = getCmduHeader();
+
+    if (cmduhdr->is_finalized())
+        swap_16((uint16_t &)mid);
```

Comment:
Is that a c-style cast (I know there are others in the file...)
Also, if you make setMessageId accept a reference instead, you don't need any cast.

Reply:
I fixed it and got rid of the other C style casts in this file.